librz/core/disasm.c

```diff
@@ -1131,6 +1142,85 @@ static void ds_build_op_str(RDisasmState *ds, bool print_color) {
 				core->parser->subrel_addr = killme;
 			}
 		}
+
+		// add for str issue
+		if (begin_mark && markinst_num > 1 && strstr(ds->opstr, "0x")) {
```

Comment:
you can use [rz_num_is_valid_input](https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_util/rz_num.h#L75)